pennylane/qnodes/base.py

```diff
@@ -525,6 +554,10 @@ def _construct(self, args, kwargs):
         # generate the DAG
         self.circuit = CircuitGraph(self.ops, self.variable_deps)
 
+        # The qfunc call may fail for various reasons.
```

Comment:
What could be reasons for failure up until this point?

Reply:
The qfunc could have a programming error in it, raising an exception, or the qfunc could fail `_check_circuit`. The user can catch and ignore the resulting exception using a `try` block. Johannes had a relevant example of this, but I can't find it right now.

Setting `last_aux_args` only here should make sure its contents match with `self.circuit`.

Reply:
```python
@qml.qnode(dev)
def qfunc(weights):
    qml.RX(weights[0], wires=[0])
    qml.RX(weights[1], wires=[1])
    qml.CRY(weights[2], wires=[0,1])

    return qml.expval(qml.PauliZ(0))

try:
    qfunc([0.2, 0.4]) # fails because weights[2] is not present
except:
    print("oops")

qfunc([0.2, 0.4, 0.6]) # Works now because we cleared the cache, otherwise it would fail here because a length-2 list would be assumed
```

Reply:
Thanks! :)